subsys/modbus/Kconfig

```diff
@@ -44,7 +44,7 @@ config MODBUS_SERIAL
 	default y
 	depends on SERIAL && SERIAL_HAS_DRIVER
 	depends on DT_HAS_ZEPHYR_MODBUS_SERIAL_ENABLED
-	select UART_USE_RUNTIME_CONFIGURE
+	imply UART_USE_RUNTIME_CONFIGURE
```

Comment:
Hi, I would remove this select/imply entirely :) The default value of `UART_USE_RUNTIME_CONFIGURE` is `y`,  so unless the user explicitly adds `CONFIG_UART_USE_RUNTIME_CONFIGURE=n` to their .prj somewhere, it will be selected. This also means that no current users of the `modbus_serial.c` have explicitly disabled runtime as it produces this Kconfig error to do so:
```
warning: UART_USE_RUNTIME_CONFIGURE (defined at drivers/serial/Kconfig:51) was assigned the value
'n' but got the value 'y'. See
http://docs.zephyrproject.org/latest/kconfig.html#CONFIG_UART_USE_RUNTIME_CONFIGURE and/or look up
UART_USE_RUNTIME_CONFIGURE in the menuconfig/guiconfig interface. The Application Development
Primer, Setting Configuration Values, and Kconfig - Tips and Best Practices sections of the manual
might be helpful too.
```

If it is possible to use this driver without `UART_USE_RUNTIME_CONFIGURE` as it is after this PR, it shall not be selected in Kconfig. `select` is explicitly used only when the dependency is "small" but obligatory :)

Reply:
I agree. The first version of this PR removed the select line entirely, but Johann refused it.

Reply:
@jfischer-no can we agree on removing the `select` not being a breaking change? or have I missed something? :)

Reply:
No.